client-hc/src/main/java/com/graphhopper/api/GHMatrixAbstractRequester.java

```diff
@@ -106,13 +106,13 @@ protected JsonNode createPostRequest(GHMRequest ghRequest, Collection<String> ou
         // requestJson.put("elevation", ghRequest.getHints().getBool("elevation", false));
         requestJson.put("fail_fast", ghRequest.getFailFast());
 
-        Map<String, String> hintsMap = ghRequest.getHints().toMap();
+        Map<String, Object> hintsMap = ghRequest.getHints().toMap();
         for (String hintKey : hintsMap.keySet()) {
             if (ignoreSet.contains(hintKey))
                 continue;
 
-            String hint = hintsMap.get(hintKey);
-            requestJson.put(hintKey, hint);
+            Object hint = hintsMap.get(hintKey);
+            requestJson.put(hintKey, hint.toString());
```

Comment:
Why do we have to turn this into a string here? Can we not just keep the object? Aren't there possible cases where we want to add the actual object to the json tree instead of converting it into a string? Hm, but this requires object-to-json serialization to create a (sub)JsonNode?